### PR TITLE
Mon 3801 csv export acl

### PR DESCRIPTION
**Pull request by sc979**
<h1> Pull Request Template </h1>

<h2> Description </h2>

Fix ACL calculation when interfering with the GET request

Fixes # (support)

<h2> Type of change </h2>

- [x] Patch fixing an issue (non-breaking change)

<h2> Target serie </h2>

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x (master)

<h2> How this pull request can be tested ? </h2>

For more details, please ask me in private

<h2> Checklist </h2>

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests covers 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php [hunk 1]
```diff
@@ -61,10 +59,13 @@
...
 
 $sid = session_id();
+
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->query(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = '" . $pearDB->escape($oreon->user->user_id) . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php [hunk 2]
```diff
@@ -74,114 +75,162 @@
...
+$hostgroupId = filter_var(
+    $_GET["hostgroup"] ?? $_POST['hostgroup'] ?? null,
+    FILTER_VALIDATE_INT
+);
 
-/*
- * Getting time interval to report
- */
+// finding the user's allowed hostgroups
+$hgs = $centreon->user->access->getHostGroupAclConf(null, 'broker');
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php]: $hgs ?
You can be more explicit with the name of the variable

File: www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php [hunk 3]
```diff
@@ -74,114 +75,162 @@
...
+        '<b>You are not allowed to access this host group</b></div>';
+    exit();
+}
+
+// Getting time interval to report
 $dates = getPeriodToReport();
 $start_date = htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date = htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$hostgroup_name = getHostgroupNameFromId($id);
+$hostgroup_name = getHostgroupNameFromId($hostgroupId);
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php]: CamelCalse mandatory

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php [hunk 4]
```diff
@@ -62,111 +60,175 @@
...
+        '<b>You are not allowed to access this host</b></div>';
+    exit();
+}
+
+// Getting time interval to report
 $dates = getPeriodToReport();
 $start_date = htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date = htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$host_name = getHostNameFromId($id);
+$host_name = getHostNameFromId($hostId);
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: CamelCase mandatory

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 5]
```diff
@@ -74,16 +74,31 @@
...
 
-isset($_GET["servicegroup"]) ? $id = htmlentities($_GET["servicegroup"], ENT_QUOTES, "UTF-8") : $id = "NULL";
-isset($_POST["servicegroup"]) ? $id = htmlentities($_POST["servicegroup"], ENT_QUOTES, "UTF-8") : $id = $id;
+$servicegroupId = filter_var(
+    $_GET['servicegroup'] ?? $_POST['servicegroup'] ?? null,
+    FILTER_VALIDATE_INT
+);
+
+// finding the user's allowed servicegroup
+$sgs = $centreon->user->access->getServiceGroupAclConf(null, 'broker');
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: You can be more explicit with the name of the variable

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 6]
```diff
@@ -74,16 +74,31 @@
...
+}
 
 /*
  * Getting time interval to report
  */
 $dates = getPeriodToReport();
 $start_date = htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date = htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$servicegroup_name = getServiceGroupNameFromId($id);
+$servicegroup_name = getServiceGroupNameFromId($servicegroupId);
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: CamelCase mandatory

File: www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php [hunk 7]
```diff
@@ -74,114 +75,162 @@
...
- */
+
+// getting all hosts from hostgroup
 $str = "";
-$request = "SELECT host_host_id FROM `hostgroup_relation` WHERE `hostgroup_hg_id` = '" .$id."'";
-$DBRESULT = $pearDB->query($request);
-while ($hg = $DBRESULT->fetchRow()) {
+$dbResult = $pearDB->query(
+    "SELECT host_host_id FROM `hostgroup_relation` " .
+    "WHERE `hostgroup_hg_id` = '" . $hostgroupId . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php [hunk 8]
```diff
@@ -62,111 +60,175 @@
...
- */
+// Checking session
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->query(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = '" . $pearDB->escape($oreon->user->user_id) . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php [hunk 9]
```diff
@@ -62,111 +60,175 @@
...
-        "AND `date_start` >= '".$start_date."' " .
-        "AND `date_end` <= '".$end_date."' " .
-        "ORDER BY `date_start` desc";
-$DBRESULT = $pearDBO->query($rq);
-
-while ($row = $DBRESULT->fetchRow()) {
+
+$dbResult = $pearDBO->query(
+    "SELECT  * FROM `log_archive_host` " .
+    "WHERE `host_id` = '" . $hostId . "' " .
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 10]
```diff
@@ -63,8 +61,10 @@
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->query(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = '" . $pearDB->escape($oreon->user->user_id) . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 11]
```diff
@@ -108,102 +129,158 @@
...
+    . $stringWarning . " " . $stringAlert . ";"
+    . $stringUnknown . " " . $stringMeanTime . ";"
+    . $stringUnknown . " " . $stringAlert . ";"
+    . $stringCritical . " " . $stringMeanTime . ";"
+    . $stringCritical . " " . $stringAlert . ";"
     . _("Day") . "\n";
 
+$dbResult = $pearDB->query(
+    "SELECT `service_service_id` FROM `servicegroup_relation` " .
+    "WHERE `servicegroup_sg_id` = '" . $servicegroupId . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 12]
```diff
@@ -108,102 +129,158 @@
...
+    "SELECT `date_start`, `date_end`, sum(`OKnbEvent`) as OKnbEvent, "
+    . "sum(`CRITICALnbEvent`) as CRITICALnbEvent, "
+    . "sum(`WARNINGnbEvent`) as WARNINGnbEvent, "
+    . "sum(`UNKNOWNnbEvent`) as UNKNOWNnbEvent, "
+    . "avg( `OKTimeScheduled` ) as OKTimeScheduled, "
+    . "avg( `WARNINGTimeScheduled` ) as WARNINGTimeScheduled, "
+    . "avg( `UNKNOWNTimeScheduled` ) as UNKNOWNTimeScheduled, "
+    . "avg( `CRITICALTimeScheduled` ) as CRITICALTimeScheduled "
+    . "FROM `log_archive_service` WHERE `service_id` IN (" . $str . ") "
+    . "AND `date_start` >= '" . $start_date . "' "
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php [hunk 13]
```diff
@@ -46,17 +46,17 @@
...
 $pearDBO = new CentreonDB("centstorage");
 
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->query(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = '" . $pearDB->escape($oreon->user->user_id) . "'"
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php [hunk 14]
```diff
@@ -66,73 +66,131 @@
...
 $dates = getPeriodToReport();
 $start_date =  htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date =  htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$host_name = getHostNameFromId($host_id);
-$service_description = getServiceDescriptionFromId($service_id);
-
-/*
- * file type setting
- */
+$host_name = getHostNameFromId($hostId);
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: CamelCase mandatory

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php [hunk 15]
```diff
@@ -66,73 +66,131 @@
...
+    . _("Critical") . " (s); "
+    . _("Critical") . " %;"
+    . _("Critical") . " Alert;"
+    . _("Day") . ";\n";
+
+$dbResult = $pearDBO->query(
+    "SELECT  * FROM `log_archive_service` " .
+    "WHERE `host_id` = '" . $hostId . "' " .
+    "AND `service_id` = '" . $serviceId . "' " .
+    "AND `date_start` >= '" . $start_date . "' " .
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: In this case, using a prepare statement is hightly recommended

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php (hunk 14 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php (hunk 8 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: Refacto the whole code is out of my scope...  ;)

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php (hunk 4 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php (hunk 9 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: Refacto the whole code is out of my scope... ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php (hunk 10 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: Refacto the whole code is out of my scope... ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php (hunk 6 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php (hunk 11 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: Refacto the whole code is out of my scope...  ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php (hunk 12 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: Refacto the whole code is out of my scope...  ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php (hunk 13 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: Refacto the whole code is out of my scope...  ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php (hunk 14 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php (hunk 15 above)
Comment (sc979) [reply www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: Refacto the whole code is out of my scope...  ;)

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php [hunk 16]
```diff
@@ -62,111 +60,181 @@
...
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->prepare(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = :id"
+    );
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
     if (!$res->rowCount()) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: rowCount works only for INSERT, UPDATE or DELETE requests.

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 17]
```diff
@@ -63,8 +61,12 @@
...
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->prepare(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = :id"
+    );
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
     if (!$res->rowCount()) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: rowCount works only for INSERT, UPDATE or DELETE requests.

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php [hunk 18]
```diff
@@ -46,17 +46,20 @@
...
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
+    $res = $pearDB->prepare(
+        "SELECT user_id FROM session " .
+        "WHERE user_id = :id"
+    );
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
+
     if (!$res->rowCount()) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: rowCount works only for INSERT, UPDATE or DELETE requests.

File: www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php [hunk 19]
```diff
@@ -61,11 +59,13 @@
...
+
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
-    if (!$res->rowCount()) {
+    $res = $pearDB->prepare("SELECT COUNT(*) FROM session WHERE user_id = :id");
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
+    if (!$res->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostGroupLogs.php]: With a "SELECT COUNT(*)" request, the fetch method will always return a value.

File: www/include/reporting/dashboard/csvExport/csv_HostLogs.php [hunk 20]
```diff
@@ -62,111 +60,178 @@
...
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
-    if (!$res->rowCount()) {
+    $res = $pearDB->prepare("SELECT COUNT(*) FROM session WHERE user_id = :id");
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
+    if (!$res->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_HostLogs.php]: With a "SELECT COUNT(*)" request, the fetch method will always return a value.

File: www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php [hunk 21]
```diff
@@ -63,9 +61,10 @@
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
-    if (!$res->rowCount()) {
+    $res = $pearDB->prepare("SELECT COUNT(*) FROM session WHERE user_id = :id");
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
+    if (!$res->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php]: With a "SELECT COUNT(*)" request, the fetch method will always return a value.

File: www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php [hunk 22]
```diff
@@ -46,18 +46,17 @@
...
 $sid = session_id();
 if (!empty($sid) && isset($_SESSION['centreon'])) {
     $oreon = $_SESSION['centreon'];
-    $query = "SELECT user_id FROM session WHERE user_id = '".$pearDB->escape($oreon->user->user_id)."'";
-    $res = $pearDB->query($query);
-    if (!$res->rowCount()) {
+    $res = $pearDB->prepare("SELECT COUNT(*) FROM session WHERE user_id = :id");
+    $res->bindValue(':id', (int)$oreon->user->user_id, PDO::PARAM_INT);
+    $res->execute();
+    if (!$res->fetch(\PDO::FETCH_ASSOC)) {
```
Comment (callapa) [inline www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php]: With a "SELECT COUNT(*)" request, the fetch method will always return a value.

**Pull request merged**